### ISSUE TITLE
`@SecureField` in members of the response class isn't applied

### ISSUE THREAD

**Issue by IsSkyfalls** [quarkusio/quarkus#39513]
Consider the following relationship
```
Shop -> {
   Owner
}
Owner -> {
    name,
    @SecureField email
}
```
When a resource returns `Shop`, the following code in the processor checks for `@SecureField` on `Shop`, but not on `Owner`(its class members). Because of this, `SecurityCustomSerialization` is not applied.

https://github.com/quarkusio/quarkus/blob/097638bfc6ea020157e1a842a04138b482c82400/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java#L426-L442

I personally think that `@SecureField` should propagate up the reference chain and apply to all classes that use the secured class. The current "flat" implementation is only sutable for response/dto classes that have `@SecureField` on its class members, which isn't always the case. Please correct me if I'm wrong about the intended use case.

Workaround:
```java
public abstract class SecureObject {
    @SecureField(rolesAllowed = "NOT_ALLOWED")
    private boolean secureFieldActivationWorkaround=false;
}
```

**Comment by michalvavrik** [comment]
I agree it is a bug. I'll have a look next week unless @IsSkyfalls or @geoand would prefer to do that?

Or maybe there is some good reason why it works this way? I can't think of one.

**Comment by geoand** [comment]
I agree it's a bug. 

Feel free to tackle it @michalvavrik

**Comment by IsSkyfalls** [comment]
So I think there are basically 2 approaches: 1.dfs on the return type and 2. recursively tagging classes that reference a class using `@SecureField`. I went the latter because `IndexView.getKnownUsers` sounded very promising. After some testing I realized that the field types are not indexed (fields do not use CONSTANT_Class, e.g. `#33 = Utf8 Ljava/util/List;`), so this is a dead end. Since I ran out of smart ideas I'll leave this for now.
